src/backend/optimizer/plan/setrefs.c

```diff
@@ -1081,12 +1080,7 @@ set_plan_refs(PlannerInfo *root, Plan *plan, int rtoffset)
 					splan->returningLists = newRL;
 
 					/*
-					 * Set up the visible plan targetlist as being the same as
-					 * the first RETURNING list. This is for the use of
-					 * EXPLAIN; the executor won't pay any attention to the
-					 * targetlist.  We postpone this step until here so that
-					 * we don't have to do set_returning_clause_references()
-					 * twice on identical targetlists.
+					 * copy the new returningList as it may be needed later.
 					 */
 					splan->plan.targetlist = copyObject(linitial(newRL));
```

Comment:
Maybe you should free the list first?

Reply:
can't other nodes refer to it? it seems to be deleted anyway when the context is deleted.